### DISCUSSION TITLE
no visible @interface for 'FIRQuery'

### DISCUSSION THREAD

**Discussion by MicxSamonte** [invertase/react-native-firebase#4677] [General]
I'm scratching my head over this since yesterday, the build works a few days ago and I haven't changed anything related to the iOS build configs or Podfile.

```
error: no visible @interface for 'FIRQuery' declares the selector 'queryWhereFieldPath:isNotEqualTo:'
      _query = [_query queryWhereFieldPath:fieldPath isNotEqualTo:value];


error: no visible @interface for 'FIRQuery' declares the selector 'queryWhereFieldPath:notIn:'
      _query = [_query queryWhereFieldPath:fieldPath notIn:value];
```

I already tried the solutions that can be found here: https://github.com/invertase/react-native-firebase/issues/3927 and https://github.com/invertase/react-native-firebase/issues/1566 but to no avail

The issue is related to the pull req on https://github.com/invertase/react-native-firebase/pull/4474 but I've already updated my dependencies:

```
"@react-native-firebase/app": "^7.2.0",
"@react-native-firebase/auth": "^8.0.4",
"@react-native-firebase/firestore": "10.0.0",
"@react-native-firebase/messaging": "^7.1.4",
```

**Comment by mikehardy** [comment]
@MicxSamonte 

> I haven't changed anything related to the iOS build configs or Podfile.

This must not be true.

More importantly you should be receiving warnings when you attempt to install that version spread, With the version 10 of the packages we switched to locked / single number versioning for all the packages at the same time so if you are on v10.0.0 of any package you should be on v10.0.0 of all packages, but you are not, and yarn or npm will warn you of peer dependencies being wrong.

That's your root cause either way. That version of `@react-native-firebase/app` requires a version of `@react-native-firebase/app` that is also v10.0.0 which will pull in an underlying version of firebase-ios-sdk that has the symbols you need.

relevant:
- https://github.com/invertase/react-native-firebase/blob/master/packages/firestore/CHANGELOG.md#790-2020-10-30
- https://github.com/invertase/react-native-firebase/blob/master/packages/app/CHANGELOG.md#840-2020-08-26

Current is now v10.2.0, you should update all packages to the same version and rebuild. it will work. You need to check the release notes of the various packages until v10 for breaking change notes, and the release notes for v10 to make sure you handle any breaking changes there. They should be small, although iOS does go to minimum iOS version 10. If you can't handle that then you need to downgrade firestore to something compatible, but you won't receive support for it here because we only handle current versions

Hope this helps
